src/ui/public/kuery/lib/functions/__tests__/is.js

```diff
@@ -0,0 +1,128 @@
+
+  describe('is', function () {
+
+    beforeEach(ngMock.module('kibana'));
+    beforeEach(ngMock.inject(function (Private) {
+      indexPattern = Private(StubbedLogstashIndexPatternProvider);
+    }));
+
+    describe('buildNodeParams', function () {
+
+      it('should return "arguments" and "serializeStyle" params', function () {
+        const result = is.buildNodeParams('response', 200);
+        expect(result).to.only.have.keys('arguments', 'serializeStyle');
+      });
+
+      it('arguments should contain the provided fieldName and value as literals', function () {
+        const { arguments: [ fieldName, value ] } = is.buildNodeParams('response', 200);
+
+        expect(fieldName).to.have.property('type', 'literal');
+        expect(fieldName).to.have.property('value', 'response');
+
+        expect(value).to.have.property('type', 'literal');
+        expect(value).to.have.property('value', 200);
+      });
+
+      it('serializeStyle should default to "operator"', function () {
+        const { serializeStyle } = is.buildNodeParams('response', 200);
+        expect(serializeStyle).to.be('operator');
+      });
+
+    });
+
+    describe('toElasticsearchQuery', function () {
+
+      it('should return an ES match_all query when fieldName is "*" and no value is provided', function () {
+        const expected = {
+          match_all: {}
+        };
+
+        const node = nodeTypes.function.buildNode('is', '*');
+        const result = is.toElasticsearchQuery(node, indexPattern);
+        expect(_.isEqual(expected, result)).to.be(true);
+      });
+
+      it('should return an ES simple_query_string query in all fields mode when fieldName is "*"', function () {
+        const expected = {
+          simple_query_string: {
+            query: '"200"',
+            all_fields: true,
+          }
+        };
+
+        const node = nodeTypes.function.buildNode('is', '*', 200);
+        const result = is.toElasticsearchQuery(node, indexPattern);
+        expect(_.isEqual(expected, result)).to.be(true);
+      });
+
+      // See discussion about kuery escaping for background:
+      // https://github.com/elastic/kibana/pull/12624#issuecomment-312650307
+      it('should ensure the simple_query_string query is wrapped in double quotes to force a phrase search', function () {
+        const node = nodeTypes.function.buildNode('is', '*', '+response');
+        const result = is.toElasticsearchQuery(node, indexPattern);
+        expect(result.simple_query_string.query).to.be('"+response"');
+      });
+
+      it('already double quoted phrases should not get wrapped a second time', function () {
+        const node = nodeTypes.function.buildNode('is', '*', '"+response"');
+        const result = is.toElasticsearchQuery(node, indexPattern);
+        expect(result.simple_query_string.query).to.be('"+response"');
+      });
+
+      it('should return an ES exists query when no value is provided', function () {
```

Comment:
Agreed, both arguments should be required. I think this was left over from some intermediate implementation.

b73bd2bbe40e7d308273d9d44af4f9ec0bab4b91